### ISSUE TITLE
make pipelines graph jsongraph compatible

### ISSUE THREAD

**Issue by 111andre111** [elastic/logstash#11836]
At the moment it is possible to monitor logstash with graph and the metricbeat implementation uses this approach as well to export the monitoring of the pipeline to elasticsearch.

```
curl -s "localhost:9600/_node/pipelines/main?graph=true" | jq ".pipelines.main.graph"
curl -s "localhost:9600/_node/stats/pipelines/main?vertices=true"  | jq ".pipelines.main.vertices"
```

My request is to make this jsongraph compatible because that would make the pipeline compatible to Graphviz / dot Graphs:
http://jsongraphformat.info
https://www.graphviz.org/doc/info/output.html#d:dot_json

If you have a compatible jsongraph, you can do lots of things and even convert it in the terminal to a PNG file and visualize it like that easily:
```
curl -s https://raw.githubusercontent.com/jsongraph/jgf-dot/master/example/graph.json | jgfdot | dot -Tpng | imgcat
```
This outputs a graph like that:
![graph](https://user-images.githubusercontent.com/48132449/80244483-4412eb00-8669-11ea-8487-aae716f0cb4c.png)

The issue here is that the actual implemention of logstash is incompatible with that because of following reasons:
    graph implementation depends on these node types:
      directed, type, metadata
      nodes
      edges
    logstash implementation is like that:
    graph
      hash, type=lir, version
      vertices
      edges

So basically the metadata information is missing and we have vertices in logstash and not the nodes. The edges are existing in both.
